erpnext/accounts/doctype/asset/asset.py

```diff
@@ -80,7 +80,7 @@ def validate_asset_values(self):
 			if cint(self.number_of_depreciations_booked) > cint(self.total_number_of_depreciations):
 				frappe.throw(_("Number of Depreciations Booked cannot be greater than Total Number of Depreciations"))
 						
-		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(nowdate()):
+		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(self.purchase_date):
```

Comment:
This is a warning message when user selects the next depreciation date before today. It does not stop to create /submit the asset.

You can add another validation for purchase date, which should block saving the asset.